Gutenberg/cocoapods_helpers.rb

```diff
@@ -80,20 +64,111 @@ def gutenberg_pod(config: GUTENBERG_CONFIG)
-  end
+  # When referencing via a tag or commit, we download pre-built frameworks.
+  return if options.key?(:tag) || options.key?(:commit)
+
+  podspec_prefix = options[:path]
+  gutenberg_path = options[:path]
+
+  raise "Unexpected Gutenberg dependencies configuration '#{options}'" if podspec_prefix.nil?
 
   podspec_prefix += '/third-party-podspecs'
   podspec_extension = 'podspec.json'
 
-  # FBReactNativeSpec needs special treatment because of react-native-codegen code generation
-  pod 'FBReactNativeSpec', podspec: "#{podspec_prefix}/FBReactNativeSpec/FBReactNativeSpec.#{podspec_extension}"
+  computed_dependencies = DEPENDENCIES.dup
 
-  DEPENDENCIES.each do |pod_name|
+  react_native_version = react_native_version!(gutenberg_path: gutenberg_path)
+  # We need to apply a workaround for the RNReanimated library when using React Native 0.71+.
+  apply_rnreanimated_workaround!(dependencies: computed_dependencies, gutenberg_path: gutenberg_path) unless react_native_version[1] < 71
+
+  computed_dependencies.each do |pod_name|
     pod pod_name, podspec: "#{podspec_prefix}/#{pod_name}.#{podspec_extension}"
   end
 end
+
+def apply_rnreanimated_workaround!(dependencies:, gutenberg_path:)
+  # RNReanimated needs React-jsc
+  # Reference: https://github.com/WordPress/gutenberg/pull/51386/commits/9538f8eaf73dfacef17382e1ab7feda40231061f
+  dependencies.push('React-jsc')
+
+  # Use a custom RNReanimated version while we coordinate a fix upstream
+  dependencies.delete('RNReanimated')
+
+  # This is required to workaround an issue with RNReanimated after upgrading to version 2.17.0
+  rn_node_modules = File.join(gutenberg_path, '..', 'gutenberg', 'node_modules')
+  raise "Could not find node modules at given path #{rn_node_modules}" unless File.exist? rn_node_modules
+
+  ENV['REACT_NATIVE_NODE_MODULES_DIR'] = rn_node_modules
+  puts "[Gutenberg] Set REACT_NATIVE_NODE_MODULES_DIR env var for RNReanimated to #{rn_node_modules}"
+
+  pod 'RNReanimated', git: 'https://github.com/wordpress-mobile/react-native-reanimated', branch: 'wp-fork-2.17.0'
+end
+
+def gutenberg_post_install(installer:)
+  return unless should_use_local_gutenberg
+
+  raise "[Gutenberg] Could not find local Gutenberg at given path #{local_gutenberg_path}" unless File.exist?(local_gutenberg_path)
+
+  react_native_path = require_react_native_helpers!(gutenberg_path: local_gutenberg_path)
+
+  puts "[Gutenberg] Running Gutenberg post install hook (RN path: #{react_native_path})"
+
+  # It seems like React Native prepends $PWD to the path internally in the post install hook.
+  # When using an absolute path, we get this error, notice the duplicated "/Users/gio/Developer/a8c/wpios":
+  #
+  #   [!] An error occurred while processing the post-install hook of the Podfile.
+  #
+  #   No such file or directory @ rb_sysopen - <GUTENBERG_MOBILE_PROJECT_PATH>/gutenberg/node_modules/react-native/package.json
+  #
```

Comment:
If we confirmed that React Native preprends `$PWD`, I think we don't need to share details about the error. Hence, we could consider removing this part.

Reply:
I've applied this suggestion via https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/commits/7e11b1e1a895173a05197dce99d6bbffdf43908f.